pkg/scheduler/plugins/loadaware/load_aware.go

```diff
@@ -120,39 +125,83 @@ func (p *Plugin) Filter(ctx context.Context, state *framework.CycleState, pod *c
 		}
 	}
 
-	usageThresholds := p.args.UsageThresholds
+	usageThresholds, prodUsageThresholds := p.args.UsageThresholds, p.args.ProdUsageThresholds
 	customUsageThresholds, err := extension.GetCustomUsageThresholds(node)
 	if err != nil {
 		klog.V(5).ErrorS(err, "failed to GetCustomUsageThresholds from", "node", node.Name)
 	} else {
 		if len(customUsageThresholds.UsageThresholds) > 0 {
 			usageThresholds = customUsageThresholds.UsageThresholds
 		}
+		if len(customUsageThresholds.ProdUsageThresholds) > 0 {
+			prodUsageThresholds = customUsageThresholds.ProdUsageThresholds
+		}
 	}
 
 	if len(usageThresholds) > 0 {
-		if nodeMetric.Status.NodeMetric == nil {
-			return nil
+		status := p.filterNodeUsage(node, nodeMetric, usageThresholds)
+		if !status.IsSuccess() {
+			return status
 		}
-		for resourceName, threshold := range usageThresholds {
-			if threshold == 0 {
-				continue
-			}
-			total := node.Status.Allocatable[resourceName]
-			if total.IsZero() {
-				continue
-			}
-			used := nodeMetric.Status.NodeMetric.NodeUsage.ResourceList[resourceName]
-			usage := int64(math.Round(float64(used.MilliValue()) / float64(total.MilliValue()) * 100))
-			if usage >= threshold {
-				return framework.NewStatus(framework.Unschedulable, fmt.Sprintf(ErrReasonUsageExceedThreshold, resourceName))
-			}
+	}
+
+	if len(prodUsageThresholds) > 0 {
+		status := p.filterProdUsage(node, nodeMetric, prodUsageThresholds)
+		if !status.IsSuccess() {
+			return status
 		}
 	}
 
 	return nil
 }
 
+func (p *Plugin) filterNodeUsage(node *corev1.Node, nodeMetric *slov1alpha1.NodeMetric, usageThresholds map[corev1.ResourceName]int64) *framework.Status {
+	if nodeMetric.Status.NodeMetric == nil {
+		return nil
+	}
+	for resourceName, threshold := range usageThresholds {
+		if threshold == 0 {
+			continue
+		}
+		total := node.Status.Allocatable[resourceName]
+		if total.IsZero() {
+			continue
+		}
+		// TODO(joseph): maybe we should estimate the Pod that just be scheduled that have not reported
+		used := nodeMetric.Status.NodeMetric.NodeUsage.ResourceList[resourceName]
+		usage := int64(math.Round(float64(used.MilliValue()) / float64(total.MilliValue()) * 100))
+		if usage >= threshold {
+			return framework.NewStatus(framework.Unschedulable, fmt.Sprintf(ErrReasonUsageExceedThreshold, resourceName))
+		}
+	}
+	return nil
+}
+
+func (p *Plugin) filterProdUsage(node *corev1.Node, nodeMetric *slov1alpha1.NodeMetric, prodUsageThresholds map[corev1.ResourceName]int64) *framework.Status {
```

Comment:
do we need count the system usage either, as some system processes or non-containerized apps can affect Prod pods?

Reply:
Actually it is needed. But this needs to extend NodeMetric CRD and be supported by koordlet. I think this extension can be used as a subsequent optimization item and implemented when it is really needed.

Reply:
OK. Currently, there is an approximation where we can subtract all pods' usage from the node usage to estimate the system usage.

Reply:
Yes. Change the NodeMetric CRD will be planned in the future